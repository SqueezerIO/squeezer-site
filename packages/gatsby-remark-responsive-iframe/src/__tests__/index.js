const Remark = require(`remark`)
const visit = require(`unist-util-visit`)

const plugin = require(`../`)

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
})

describe(`gatsby-remark-responsive-iframe`, () => {
  ;[`iframe`, `object`].forEach(tag => {
    it(`transforms an ${tag}`, async () => {
      const markdownAST = remark.parse(`
<${tag} url="http://www.example.com/" width="600" height="400"></${tag}>
    `)

      const transformed = await plugin({ markdownAST })

      visit(transformed, `unknown`, node => {
        expect(node.data.hChildren.length).toBeGreaterThan(0)
        const [child] = node.data.hChildren
        expect(child.value).toMatchSnapshot()
      })
    })
  })
})
