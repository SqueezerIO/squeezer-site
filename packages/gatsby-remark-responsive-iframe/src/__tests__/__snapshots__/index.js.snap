// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-responsive-iframe transforms an iframe 1`] = `
"
          <div
            class=\\"gatsby-resp-iframe-wrapper\\"
            style=\\"padding-bottom: 66.66666666666666%; position: relative; height: 0; overflow: hidden;\\"
          >
            <html><head></head><body><iframe url=\\"http://www.example.com/\\" style=\\"
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
          \\"></iframe>
    </body></html>
          </div>
          "
`;

exports[`gatsby-remark-responsive-iframe transforms an object 1`] = `
"
          <div
            class=\\"gatsby-resp-iframe-wrapper\\"
            style=\\"padding-bottom: 66.66666666666666%; position: relative; height: 0; overflow: hidden;\\"
          >
            <html><head></head><body><object url=\\"http://www.example.com/\\" style=\\"
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
          \\"></object></body></html>
          </div>
          "
`;
